src/ecc_key.c

```diff
@@ -943,9 +1033,30 @@ int zpc_ec_key_reencipher(struct zpc_ec_key *ec_key, unsigned int method)
 			if (rc)
 				continue;
 
+			/* Note that the secure key is a TOKVER_EP11_ECC_WITH_HEADER and has a
+			 * 16-byte ep11kblob_header prepended before the actual secure key blob.
+			 * For reencipher we have to skip this prepended hdr and provide the
+			 * ep11kblob_header info as an overlay over the session field at
+			 * the beginning of the secure key. So at this point we assume that
+			 * the key's session id field does not contain any info! */
+			ep11hdr = (struct ep11kblob_header *)((unsigned char *)reenc.sec +
```

Comment:
Please check if you really overlay the session field here. 
I would strongly suggest to ALWAYS use the EP11 key with header IN FRONT, and not the overlayed one.

Reply:
I guess this still needs to be fixed to not use an overlay of the session field?

Reply:
Fixed. Code is reworked.